### PR TITLE
Reverse Linked List problem solution

### PR DESCRIPTION
**Pull request by makcgr**

### CODE REVIEW

File: mgrachev/reverse linked list/Program.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+{
+    public static Node ReverseLinkedList(Node node)
+    {
+        if (node == null) return null;
+        
+        Node prev = null;
+        Node cur = node;
+        while (cur != null)
+        {
+            var memoized = cur.Next;
```
Comment (vlemish) [inline mgrachev/reverse linked list/Program.cs:72]: you made me to google what `memoized` is 🙂 Now I know :D

Comment (makcgr) [reply mgrachev/reverse linked list/Program.cs:72]: Yeah, it's a neat word, right? 😃

Comment (vlemish) [reply mgrachev/reverse linked list/Program.cs:72]: I've never seen before :D

File: mgrachev/reverse linked list/Program.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+One for  "previously" visited node (prev), and one for "current" node (cur). 
+
+We will start with prev = null and cur = first node.
+
+     (1) -> (2) -> (3) -> (4) -> (5) ->
+^     ^
+|     |
+prev  cur
+
+On each iteration step we will retarget "current" node onto the "previous" node,
```
Comment (PShchegolevatykh) [inline mgrachev/reverse linked list/Program.cs:31]: Such an elegant algorithm :) and thanks as usual for detailed descriptions, really helps for noobs like me to immediately understand what's happening.

File: mgrachev/maximum depth of binary tree/Program.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        return DepthSearch(root, 0);
+    }
+
+    private static int DepthSearch(TreeNode node, int curDepth)
+    {
+        if (node == null) return curDepth;
+        curDepth++;
+
+        return Math.Max(DepthSearch(node.Left, curDepth), DepthSearch(node.Right, curDepth));
```
Comment (PShchegolevatykh) [inline mgrachev/maximum depth of binary tree/Program.cs:42]: Neat :) backfilling missed assignments.

**Review by PShchegolevatykh — APPROVED** [review]

**Pull request merged**